Makefile

```diff
@@ -29,7 +39,7 @@ clean:
 # remove running and stopped test containers
 	-@ [[ -d config.bak ]] && { rm -rf config ; mv config.bak config ; } || :
 	-@ [[ -d testconfig.bak ]] && { sudo rm -rf test/config ; mv testconfig.bak test/config ; } || :
-	-@ docker ps -a | grep -E "mail|ldap_for_mail|mail_overri.*" | cut -f 1-1 -d ' ' | xargs --no-run-if-empty docker rm -f
+	-@ for cn in $$(grep -hr ".*docker run.*-name" ./* | xargs | sed 's/.*docker run.*-name//g'); do docker rm -f "$$cn" || true; done &>/dev/null
```

Comment:
I'm not sure about this change. The previous content was clear that it was for containers with specific names, but I'm not proficient enough with shell scripts to decipher how this loop is getting container names to remove?

There is no `docker ps` involved, it looks like some input source is parsed for `--name` options on `docker run` commands?

Reply:
```
    ~/docker-mailserver  on   macos-lint-and-test-support  for cn in $(grep -hr ".*docker run.*-name" ./* | xargs | sed 's/.*docker run.*-name//g'); do echo "$cn" || true; done 
mail_default_dhparams_one_dir
mail_default_dhparams_not_one_dir
mail_postfix_inet_default
mail_postfix_inet_all
mail_postfix_inet_ipv4
mail_postfix_inet_ipv6
mail_lmtp_ip
mail_manual_dhparams_not_one_dir
mail_pop3
mail_smtponly
mail_undef_spam_subject
mail_privacy
mail_fetchmail
mail_spam_bounced_defined
mail_spam_bounced_undefined
mail_fail2ban
fail-auth-mailer
mail_manual_dhparams_one_dir
mail_smtponly_second_network_sender
mail_fetchmail_parallel
mail_disabled_clamav_spamassassin
mail_spam_moved_junk
mail_spam_moved_new
mail_srs_domainname
mail_domainname
mail_with_default_relay
mail_postscreen
mail_postscreen_sender
mail_with_imap
mail_lets_domain
mail_lets_hostname
mail_lets_acme_json
ldap_for_mail
mail_with_ldap
mail
mail_with_relays
mail_special_use_folders
mail_with_mdbox_format
tls_test_cipherlists
mail_with_postgrey
mail_with_sdbox_format
mail_override_hostname
mail_no_quotas
mail_manual_ssl
```

```
    ~/docker-mailserver  on   macos-lint-and-test-support  grep -rl ".*docker run.*-name" ./*               2 ✘  at 21:50:33  
./Makefile
./test/mail_dhparams_default.bats
./test/mail_postfix_inet.bats
./test/mail_lmtp_ip.bats
./test/mail_dhparams_manual_not_one_dir.bats
./test/mail_pop3.bats
./test/mail_smtponly.bats
./test/mail_undef_spam_subject.bats
./test/mail_privacy.bats
./test/mail_fetchmail.bats
./test/mail_spam_bounced.bats
./test/mail_fail2ban.bats
./test/mail_dhparams_manual_one_dir.bats
./test/permit_docker.bats
./test/mail_fetchmail_parallel.bats
./test/mail_disabled_clamav_spamassassin.bats
./test/mail_spam_junk_folder.bats
./test/mail_srs_domainname.bats
./test/default_relay_host.bats
./test/mail_postscreen.bats
./test/mail_with_imap.bats
./test/mail_ssl_letsencrypt.bats
./test/mail_with_ldap.bats
./test/tests.bats
./test/mail_with_relays.bats
./test/mail_special_use_folders.bats
./test/mail_with_mdbox.bats
./test/security_tls_cipherlists.bats
./test/mail_with_postgrey.bats
./test/mail_with_sdbox.bats
./test/mail_override_hostname.bats
./test/mail_quotas_disabled.bats
./test/mail_ssl_manual.bats
```

This loop is futureproofing and eliminating the need for a human to remember to add the specific name/pattern to the old grep in the Makefile.

Reply:
That needs to be explained via comment, I don't know how many are going to know what to expect from that line, whereas the previous one was far more clear with what was going on.

While neat, by being implicit like this it's far easier to gloss over that list and assume it's working correctly... but not all tests are adhering to such a convention to match on, eg [`open_dkim.bats`](https://github.com/docker-mailserver/docker-mailserver/blob/master/test/open_dkim.bats), [`permit_docker.bats`](https://github.com/docker-mailserver/docker-mailserver/blob/master/test/permit_docker.bats#L12), [this one doesn't even assign a name](https://github.com/docker-mailserver/docker-mailserver/blob/master/test/mail_with_postgrey_disabled_by_default.bats), etc.

While probably not entirely foolproof either, I believe we have a common image name at least for instances of the docker-mailserver image. There is support for customizing the image used, but querying containers using that image is probably better and more clear of intent?

I'm personally :-1: for this due to the above concern (alternatively, with the comment note included, all tests could be reviewed to conform to the convention pattern being matched and some documentation for test contributions about this convention being required so no future test contributions fail to match). I realize that the earlier approach also missed these examples I linked to, but at least it was more evident at a glance.

Reply:
Ok, I'm fine reverting it. It was just an idea I wanted to try out :)

Reply:
That's all good, thank you. At least it drew attention that we're not covering all the containers due to lack of consistent convention :sweat_smile: